tools/mpremote/mpremote/main.py

```diff
@@ -463,9 +468,21 @@ def main():
             if pyb is None:
                 pyb = do_connect(["auto"])
 
+            if cmd == "soft-reset" and not soft_reset:
+                if pyb.in_raw_repl:
+                    pyb.exit_raw_repl()
+                soft_reset = True
```

Comment:
I think the command name `soft-reset` is a bit confusing because it doesn't actually do a soft-reset.  Maybe `use-soft-reset` would be a better name.

Or actually make it do a soft reset, that might be better, like this:
```python
if pyb.in_raw_repl:
    pyb.exit_raw_repl()
pyb.enter_raw_repl()
```
Note that it wont do `soft_reset = True`, that would make the semantics hard to describe.

Or, for simplicity, just remove this command altogether and have only the `no-soft-reset` command (which should be enough for at least your original use case?).

Reply:
Having **no-soft-reset** comamnd only makes no sense in leaving it as a command because in such a case it cannot be used elsewhere than prior to first exec/eval/run command (otherwise the soft reset is already performed and it is performed only once on `enter_raw_repl()` in each comamnd chain), so I see two meaningful alternatives:

1. using `--no-soft-reset` command line option, which changes the default behavior of the command chain, i.e. no soft reset is performed on first exec/eval/run (my first commit) or
2. using commands `no-soft-reset` and `soft-reset` (renaming it to `with-soft-reset`, it' a good point). This will allow to control the soft reset behavior in more flexible way.

I would prefer the first one because it fulfills my use case (possibility to make multiple calls of exec/eval/run without soft resets). The second alternative probably makes no sense as I can't imagine any meaningful usecase for it.
I'm also thinking about another possibility:

3. Allow setting of soft_reset behavior be set in either config file or enviroment variable and allow users to change it with options or commands.

Let somone decide, I personally prefer alternative 1 ;-)